specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-04-02-preview/managedClusters.json

```diff
@@ -6083,6 +6091,23 @@
       },
       "description": "Network plugin used for building the Kubernetes network."
     },
+    "NetworkPluginMode": {
+      "type": "string",
+      "enum": [
+        "Overlay"
+      ],
+      "x-ms-enum": {
+        "name": "networkPluginMode",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "Overlay",
+            "description": "Pods are given IPs from the PodCIDR address space but use Azure Routing Domains rather than Kubenet reference plugins host-local and bridge."
+          }
+        ]
+      },
+      "description": "The mode the network plugin should use."
```

Comment:
Document the default if it's not specified? Also can this be used with only `Azure` networking mode? If so, say that.

Reply:
the default value will be "unspecified" but we weren't planning to make that an enum value. Still state it in the description?

Reply:
Oh, what does unspecified mean? Something different than overlay? If so then yea that's awkward to document... IMO in a perfect world we'd have a value that represented what the mode is "by default", but if we don't have that then yea leaving this undocumented is fine I guess.

Reply:
in the future it will most likely be "nodeSubnet" but for now overlay is the only possible plugin mode so having everything else be "unspecified" is easier to deal with.

Reply:
Yeah in that case for now having this undocumented is fine IMO. In the future when there is a representable default for everything we should document it here, but it sounds like we're not there yet.